### ISSUE TITLE
[Bug]: Error when using Mistral codestral - litellm.utils.UnsupportedParamsError: mistral does not support parameters

### ISSUE THREAD

**Issue by gcasilva** [OpenDevin/OpenDevin#2464]
### Is there an existing issue for the same bug?

- [X] I have checked the troubleshooting document at https://opendevin.github.io/OpenDevin/modules/usage/troubleshooting
- [X] I have checked the existing issues.

### Describe the bug

15:49:48 - opendevin:ERROR: agent_controller.py:137 - Traceback (most recent call last):
  File "/app/.venv/lib/python3.12/site-packages/litellm/main.py", line 816, in completion
    optional_params = get_optional_params(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 5863, in get_optional_params
    _check_valid_arg(supported_params=supported_params)
  File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 5339, in _check_valid_arg
    raise UnsupportedParamsError(
litellm.utils.UnsupportedParamsError: mistral does not support parameters: {'stop': ['</execute_ipython>', '</execute_bash>', '</execute_browse>']}, for model=codestral-latest. To drop these, set `litellm.drop_params=True` or for proxy:

`litellm_settings:
 drop_params: true`

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/app/opendevin/controller/agent_controller.py", line 130, in _start_step_loop
    await self._step()
  File "/app/opendevin/controller/agent_controller.py", line 290, in _step
    action = self.agent.step(self.state)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/agenthub/codeact_agent/codeact_agent.py", line 228, in step
    response = self.llm.do_completion(
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/opendevin/llm/llm.py", line 217, in do_completion
    resp = self._completion(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 330, in wrapped_f
    return self(f, *args, **kw)
           ^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 467, in __call__
    do = self.iter(retry_state=retry_state)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 368, in iter
    result = action(retry_state)
             ^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 390, in <lambda>
    self._add_action_func(lambda rs: rs.outcome.result())
                                     ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/_base.py", line 449, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 470, in __call__
    result = fn(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^
  File "/app/opendevin/llm/llm.py", line 197, in wrapper
    resp = completion_unwrapped(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 3415, in wrapper
    raise e
  File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 3306, in wrapper
    result = original_function(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/litellm/main.py", line 2447, in completion
    raise exception_type(
          ^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 9989, in exception_type
    raise e
  File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 8719, in exception_type
    raise APIError(
litellm.exceptions.APIError: APIError: MistralException - mistral does not support parameters: {'stop': ['</execute_ipython>', '</execute_bash>', '</execute_browse>']}, for model=codestral-latest. To drop these, set `litellm.drop_params=True` or for proxy:

`litellm_settings:
 drop_params: true`

### Current OpenDevin version

```bash
ghcr.io/opendevin/opendevin:0.6
```

### Installation and Configuration

```bash
WORKSPACE_BASE=$(pwd)/workspace
docker run -it \
    --pull=always \
    -e SANDBOX_USER_ID=$(id -u) \
    -e PERSIST_SANDBOX="true" \
    -e SSH_PASSWORD="passw0rd" \
    -e WORKSPACE_MOUNT_PATH=$WORKSPACE_BASE \
    -v $WORKSPACE_BASE:/opt/workspace_base \
    -v /var/run/docker.sock:/var/run/docker.sock \
    -p 3000:3000 \
    --add-host host.docker.internal:host-gateway \
    --name opendevin-app-$(date +%Y%m%d%H%M%S) \
    ghcr.io/opendevin/opendevin:0.6
```

### Model and Agent

- Model: Mistral codestral
- Agent: CodeActAgent

### Operating System

Mac OS

### Reproduction Steps

1. Start docker container and for prompt type: Google search Opendevin

### Logs, Errors, Screenshots, and Additional Context

2024-06-16 08:47:02 Starting OpenDevin...
2024-06-16 08:47:02 Setting up enduser with id 501
2024-06-16 08:47:02 Docker socket group id: 0
2024-06-16 08:47:02 root:x:0:
2024-06-16 08:47:02 Group with id 0 already exists
2024-06-16 08:47:04 Running as enduser
2024-06-16 08:47:12 /app/.venv/lib/python3.12/site-packages/llama_index_client/types/metadata_filter.py:20: SyntaxWarning: invalid escape sequence '\*'
2024-06-16 08:47:12   """
2024-06-16 08:47:14 There was a problem when trying to write in your cache folder (/home/enduser/.cache/huggingface/hub). You should set the environment variable TRANSFORMERS_CACHE to a writable directory.
2024-06-16 08:47:14 INFO:     Started server process [36]
2024-06-16 08:47:14 INFO:     Waiting for application startup.
2024-06-16 08:47:14 INFO:     Application startup complete.
2024-06-16 08:47:14 INFO:     Uvicorn running on http://0.0.0.0:3000 (Press CTRL+C to quit)
2024-06-16 08:47:58 INFO:     192.168.65.1:62784 - "GET /assets/index-BD1mcvjX.js HTTP/1.1" 200 OK
2024-06-16 08:47:58 INFO:     192.168.65.1:23325 - "GET /assets/index-Dns9hXiO.css HTTP/1.1" 200 OK
2024-06-16 08:47:58 INFO:     192.168.65.1:62784 - "GET /favicon-32x32.png HTTP/1.1" 304 Not Modified
2024-06-16 08:47:58 INFO:     192.168.65.1:62784 - "GET /locales/en/translation.json HTTP/1.1" 304 Not Modified
2024-06-16 08:47:58 INFO:     192.168.65.1:62784 - "GET /api/options/models HTTP/1.1" 200 OK
2024-06-16 08:47:59 INFO:     192.168.65.1:62784 - "GET /api/options/agents HTTP/1.1" 200 OK
2024-06-16 08:48:00 INFO:     ('192.168.65.1', 61461) - "WebSocket /ws" [accepted]
2024-06-16 08:48:00 INFO:     connection open
2024-06-16 08:48:00 15:48:00 - opendevin:INFO: agent.py:64 - Using server runtime
2024-06-16 08:48:00 15:48:00 - opendevin:INFO: ssh_box.py:206 - SSHBox is running as opendevin user with USER_ID=501 in the sandbox
2024-06-16 08:48:00 15:48:00 - opendevin:INFO: ssh_box.py:248 - Detected initial session.
2024-06-16 08:48:00 15:48:00 - opendevin:INFO: ssh_box.py:250 - Creating new Docker container
2024-06-16 08:48:00 15:48:00 - opendevin:WARNING: ssh_box.py:700 - Using port forwarding for Mac OS. Server started by OpenDevin will not be accessible from the host machine at the moment. See https://github.com/OpenDevin/OpenDevin/issues/897 for more information.
2024-06-16 08:48:00 15:48:00 - opendevin:INFO: ssh_box.py:674 - Mounting workspace directory: /Users/gcasilva/Documents/Personal Projects/OpenDevin/workspace
2024-06-16 08:48:00 15:48:00 - opendevin:INFO: ssh_box.py:709 - Mounting volumes: {'/Users/gcasilva/Documents/Personal Projects/OpenDevin/workspace': {'bind': '/workspace', 'mode': 'rw'}, '/tmp/cache': {'bind': '/home/opendevin/.cache', 'mode': 'rw'}}
2024-06-16 08:48:00 15:48:00 - opendevin:INFO: ssh_box.py:674 - Mounting workspace directory: /Users/gcasilva/Documents/Personal Projects/OpenDevin/workspace
2024-06-16 08:48:31 15:48:31 - opendevin:INFO: ssh_box.py:720 - Container started
2024-06-16 08:48:32 15:48:32 - opendevin:INFO: ssh_box.py:736 - waiting for container to start: 1, container status: running
2024-06-16 08:48:33 15:48:33 - opendevin:INFO: ssh_box.py:395 - Connecting to SSH session...
2024-06-16 08:48:33 15:48:33 - opendevin:INFO: ssh_box.py:397 - You can debug the SSH connection by running: `ssh -v -p 63710 opendevin@host.docker.internal` using your SSH password
2024-06-16 08:48:34 15:48:34 - opendevin:INFO: ssh_box.py:401 - Connected to SSH session
2024-06-16 08:48:35 15:48:35 - opendevin:INFO: agent.py:96 - Creating agent CodeActAgent using LLM mistral/codestral-latest
2024-06-16 08:48:35 15:48:35 - opendevin:INFO: llm.py:120 - Initializing LLM with model: mistral/codestral-latest
2024-06-16 08:48:35 15:48:35 - opendevin:INFO: mixin.py:32 - Initializing plugins in the sandbox
2024-06-16 08:48:35 15:48:35 - opendevin:INFO: mixin.py:42 - Copied files from [/app/opendevin/runtime/plugins/agent_skills] to [/opendevin/plugins/agent_skills] inside sandbox.
2024-06-16 08:48:35 15:48:35 - opendevin:INFO: mixin.py:50 - Initializing plugin [agent_skills] by executing [/opendevin/plugins/agent_skills/setup.sh] in the sandbox.
2024-06-16 08:48:42 15:48:42 - opendevin:INFO: mixin.py:67 - Plugin agent_skills initialized successfully
2024-06-16 08:48:42 15:48:42 - opendevin:INFO: mixin.py:42 - Copied files from [/app/opendevin/runtime/plugins/jupyter] to [/opendevin/plugins/jupyter] inside sandbox.
2024-06-16 08:48:42 15:48:42 - opendevin:INFO: mixin.py:50 - Initializing plugin [jupyter] by executing [/opendevin/plugins/jupyter/setup.sh] in the sandbox.
2024-06-16 08:48:52 15:48:52 - opendevin:INFO: mixin.py:67 - Plugin jupyter initialized successfully
2024-06-16 08:48:52 15:48:52 - opendevin:INFO: mixin.py:83 - Sourced ~/.bashrc successfully
2024-06-16 08:48:52 15:48:52 - opendevin:INFO: browser_env.py:43 - Starting browser env...
2024-06-16 08:48:52 15:48:52 - opendevin:ERROR: state.py:64 - Failed to restore state from session: sessions/3c9404f6-e799-4d06-8bdb-50f082ad6034/agent_state.pkl
2024-06-16 08:48:52 Error restoring state sessions/3c9404f6-e799-4d06-8bdb-50f082ad6034/agent_state.pkl
2024-06-16 08:48:52 15:48:52 - opendevin:INFO: agent_controller.py:183 - [Agent Controller 3c9404f6-e799-4d06-8bdb-50f082ad6034] Setting agent(CodeActAgent) state from AgentState.LOADING to AgentState.INIT
2024-06-16 08:48:52 15:48:52 - opendevin:INFO: agent_controller.py:127 - [Agent Controller 3c9404f6-e799-4d06-8bdb-50f082ad6034] Starting step loop...
2024-06-16 08:48:52 INFO:     192.168.65.1:29108 - "GET /api/list-files?path=/ HTTP/1.1" 200 OK
2024-06-16 08:48:57 15:48:57 - opendevin:INFO: browser_env.py:58 - Browser env started.
2024-06-16 08:49:48 15:49:48 - OBSERVATION
2024-06-16 08:49:48 **MessageAction** (source=EventSource.USER)
2024-06-16 08:49:48 CONTENT: Google search Opendevin
2024-06-16 08:49:48 15:49:48 - opendevin:INFO: agent_controller.py:183 - [Agent Controller 3c9404f6-e799-4d06-8bdb-50f082ad6034] Setting agent(CodeActAgent) state from AgentState.INIT to AgentState.RUNNING
2024-06-16 08:49:48 INFO:     192.168.65.1:41936 - "GET /api/list-files?path=/ HTTP/1.1" 200 OK
2024-06-16 08:49:48 
2024-06-16 08:49:48 
2024-06-16 08:49:48 ==============
2024-06-16 08:49:48 CodeActAgent LEVEL 0 STEP 0
2024-06-16 08:49:48 
2024-06-16 08:49:48 
2024-06-16 08:49:48 Give Feedback / Get Help: https://github.com/BerriAI/litellm/issues/new
2024-06-16 08:49:48 LiteLLM.Info: If you need to debug this error, use `litellm.set_verbose=True'.
2024-06-16 08:49:48 
2024-06-16 08:49:48 
2024-06-16 08:49:48 Provider List: https://docs.litellm.ai/docs/providers
2024-06-16 08:49:48 
2024-06-16 08:49:48 Traceback (most recent call last):
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/litellm/main.py", line 816, in completion
2024-06-16 08:49:48     optional_params = get_optional_params(
2024-06-16 08:49:48                       ^^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 5863, in get_optional_params
2024-06-16 08:49:48     _check_valid_arg(supported_params=supported_params)
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 5339, in _check_valid_arg
2024-06-16 08:49:48     raise UnsupportedParamsError(
2024-06-16 08:49:48 litellm.utils.UnsupportedParamsError: mistral does not support parameters: {'stop': ['</execute_ipython>', '</execute_bash>', '</execute_browse>']}, for model=codestral-latest. To drop these, set `litellm.drop_params=True` or for proxy:
2024-06-16 08:49:48 
2024-06-16 08:49:48 `litellm_settings:
2024-06-16 08:49:48  drop_params: true`
2024-06-16 08:49:48 
2024-06-16 08:49:48 
2024-06-16 08:49:48 During handling of the above exception, another exception occurred:
2024-06-16 08:49:48 
2024-06-16 08:49:48 Traceback (most recent call last):
2024-06-16 08:49:48   File "/app/opendevin/controller/agent_controller.py", line 130, in _start_step_loop
2024-06-16 08:49:48     await self._step()
2024-06-16 08:49:48   File "/app/opendevin/controller/agent_controller.py", line 290, in _step
2024-06-16 08:49:48     action = self.agent.step(self.state)
2024-06-16 08:49:48              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/agenthub/codeact_agent/codeact_agent.py", line 228, in step
2024-06-16 08:49:48     response = self.llm.do_completion(
2024-06-16 08:49:48                ^^^^^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/opendevin/llm/llm.py", line 217, in do_completion
2024-06-16 08:49:48     resp = self._completion(*args, **kwargs)
2024-06-16 08:49:48            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 330, in wrapped_f
2024-06-16 08:49:48     return self(f, *args, **kw)
2024-06-16 08:49:48            ^^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 467, in __call__
2024-06-16 08:49:48     do = self.iter(retry_state=retry_state)
2024-06-16 08:49:48          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 368, in iter
2024-06-16 08:49:48     result = action(retry_state)
2024-06-16 08:49:48              ^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 390, in <lambda>
2024-06-16 08:49:48     self._add_action_func(lambda rs: rs.outcome.result())
2024-06-16 08:49:48                                      ^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/usr/local/lib/python3.12/concurrent/futures/_base.py", line 449, in result
2024-06-16 08:49:48     return self.__get_result()
2024-06-16 08:49:48            ^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/usr/local/lib/python3.12/concurrent/futures/_base.py", line 401, in __get_result
2024-06-16 08:49:48     raise self._exception
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 470, in __call__
2024-06-16 08:49:48     result = fn(*args, **kwargs)
2024-06-16 08:49:48              ^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/opendevin/llm/llm.py", line 197, in wrapper
2024-06-16 08:49:48     resp = completion_unwrapped(*args, **kwargs)
2024-06-16 08:49:48            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 3415, in wrapper
2024-06-16 08:49:48     raise e
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 3306, in wrapper
2024-06-16 08:49:48     result = original_function(*args, **kwargs)
2024-06-16 08:49:48              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/litellm/main.py", line 2447, in completion
2024-06-16 08:49:48     raise exception_type(
2024-06-16 08:49:48           ^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 9989, in exception_type
2024-06-16 08:49:48     raise e
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 8719, in exception_type
2024-06-16 08:49:48     raise APIError(
2024-06-16 08:49:48 litellm.exceptions.APIError: APIError: MistralException - mistral does not support parameters: {'stop': ['</execute_ipython>', '</execute_bash>', '</execute_browse>']}, for model=codestral-latest. To drop these, set `litellm.drop_params=True` or for proxy:
2024-06-16 08:49:48 
2024-06-16 08:49:48 `litellm_settings:
2024-06-16 08:49:48  drop_params: true`
2024-06-16 08:49:48 
2024-06-16 08:49:48 15:49:48 - opendevin:ERROR: agent_controller.py:136 - Error while running the agent: APIError: MistralException - mistral does not support parameters: {'stop': ['</execute_ipython>', '</execute_bash>', '</execute_browse>']}, for model=codestral-latest. To drop these, set `litellm.drop_params=True` or for proxy:
2024-06-16 08:49:48 
2024-06-16 08:49:48 `litellm_settings:
2024-06-16 08:49:48  drop_params: true`
2024-06-16 08:49:48 
2024-06-16 08:49:48 15:49:48 - opendevin:ERROR: agent_controller.py:137 - Traceback (most recent call last):
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/litellm/main.py", line 816, in completion
2024-06-16 08:49:48     optional_params = get_optional_params(
2024-06-16 08:49:48                       ^^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 5863, in get_optional_params
2024-06-16 08:49:48     _check_valid_arg(supported_params=supported_params)
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 5339, in _check_valid_arg
2024-06-16 08:49:48     raise UnsupportedParamsError(
2024-06-16 08:49:48 litellm.utils.UnsupportedParamsError: mistral does not support parameters: {'stop': ['</execute_ipython>', '</execute_bash>', '</execute_browse>']}, for model=codestral-latest. To drop these, set `litellm.drop_params=True` or for proxy:
2024-06-16 08:49:48 
2024-06-16 08:49:48 `litellm_settings:
2024-06-16 08:49:48  drop_params: true`
2024-06-16 08:49:48 
2024-06-16 08:49:48 
2024-06-16 08:49:48 During handling of the above exception, another exception occurred:
2024-06-16 08:49:48 
2024-06-16 08:49:48 Traceback (most recent call last):
2024-06-16 08:49:48   File "/app/opendevin/controller/agent_controller.py", line 130, in _start_step_loop
2024-06-16 08:49:48     await self._step()
2024-06-16 08:49:48   File "/app/opendevin/controller/agent_controller.py", line 290, in _step
2024-06-16 08:49:48     action = self.agent.step(self.state)
2024-06-16 08:49:48              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/agenthub/codeact_agent/codeact_agent.py", line 228, in step
2024-06-16 08:49:48     response = self.llm.do_completion(
2024-06-16 08:49:48                ^^^^^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/opendevin/llm/llm.py", line 217, in do_completion
2024-06-16 08:49:48     resp = self._completion(*args, **kwargs)
2024-06-16 08:49:48            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 330, in wrapped_f
2024-06-16 08:49:48     return self(f, *args, **kw)
2024-06-16 08:49:48            ^^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 467, in __call__
2024-06-16 08:49:48     do = self.iter(retry_state=retry_state)
2024-06-16 08:49:48          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 368, in iter
2024-06-16 08:49:48     result = action(retry_state)
2024-06-16 08:49:48              ^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 390, in <lambda>
2024-06-16 08:49:48     self._add_action_func(lambda rs: rs.outcome.result())
2024-06-16 08:49:48                                      ^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/usr/local/lib/python3.12/concurrent/futures/_base.py", line 449, in result
2024-06-16 08:49:48     return self.__get_result()
2024-06-16 08:49:48            ^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/usr/local/lib/python3.12/concurrent/futures/_base.py", line 401, in __get_result
2024-06-16 08:49:48     raise self._exception
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/tenacity/__init__.py", line 470, in __call__
2024-06-16 08:49:48     result = fn(*args, **kwargs)
2024-06-16 08:49:48              ^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/opendevin/llm/llm.py", line 197, in wrapper
2024-06-16 08:49:48     resp = completion_unwrapped(*args, **kwargs)
2024-06-16 08:49:48            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 3415, in wrapper
2024-06-16 08:49:48     raise e
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 3306, in wrapper
2024-06-16 08:49:48     result = original_function(*args, **kwargs)
2024-06-16 08:49:48              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/litellm/main.py", line 2447, in completion
2024-06-16 08:49:48     raise exception_type(
2024-06-16 08:49:48           ^^^^^^^^^^^^^^^
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 9989, in exception_type
2024-06-16 08:49:48     raise e
2024-06-16 08:49:48   File "/app/.venv/lib/python3.12/site-packages/litellm/utils.py", line 8719, in exception_type
2024-06-16 08:49:48     raise APIError(
2024-06-16 08:49:48 litellm.exceptions.APIError: APIError: MistralException - mistral does not support parameters: {'stop': ['</execute_ipython>', '</execute_bash>', '</execute_browse>']}, for model=codestral-latest. To drop these, set `litellm.drop_params=True` or for proxy:
2024-06-16 08:49:48 
2024-06-16 08:49:48 `litellm_settings:
2024-06-16 08:49:48  drop_params: true`
2024-06-16 08:49:48 
2024-06-16 08:49:48 
2024-06-16 08:49:48 15:49:48 - opendevin:INFO: agent_controller.py:183 - [Agent Controller 3c9404f6-e799-4d06-8bdb-50f082ad6034] Setting agent(CodeActAgent) state from AgentState.RUNNING to AgentState.ERROR
2024-06-16 08:49:48 INFO:     192.168.65.1:41936 - "GET /api/list-files?path=/ HTTP/1.1" 200 OK

**Comment by SmartManoj** [comment]
Duplicate of https://github.com/OpenDevin/OpenDevin/issues/2421#issuecomment-2166071123

**Comment by tobitege** [comment]
@gcasilva this is a provider-dependent issue, whether they support parameters or not.
Judging from litellm's docs, these are providers that would support the parameter in question here:
The following providers explicitly support the [stop](file:///d%3A/github/OpenDevin/opendevin/llm/llm.py#184%2C13-184%2C13) parameter:

* Anthropic
* OpenAI
* Azure OpenAI
* Cohere
* Huggingface
* Openrouter
* AI21 

Can you confirm, that your provider is none of the above, please?

**Comment by gcasilva** [comment]
> @gcasilva this is a provider-dependent issue, whether they support parameters or not - and specifically the "stop" parameter as per your logs.
> 
> Judging from litellm's docs, these are providers that would support the parameter in question here: The following providers explicitly support the stop parameter:
> 
> * Anthropic
> * OpenAI
> * Azure OpenAI
> * Cohere
> * Huggingface
> * Openrouter
> * AI21
> 
> Can you confirm, that your provider is none of the above, please?

Provider is Mistral. So does that mean Opendevin doesn't support providers that don't support the stop parameter?

**Comment by tobitege** [comment]
It is a fixable, technical issue currently, it is certainly not intentional.
I already have a fix for this in my head, just need to get it implemented, tested etc.

Please stay tuned and thanks for the provided, invaluable logs for assessment!

**Comment by SmartManoj** [comment]
Could you pass `-e LITELLM_DROP_PARAMS="true"`?

https://github.com/BerriAI/litellm/blob/3a35a58859a145a4a568548316a1930340e7440a/litellm/__init__.py#L76

**Comment by tobitege** [comment]
Fixed by #2471
